File: HomeworkCPP07/Task2.cpp
// Задача 02
// Напишете функция, която приема вектор от цели числа и проверява дали съществува под-вектор (последователност
// от елементи в оригиналния контейнер), чиято сума на елементите е равна на 0.

#include <vector>
bool equalsSequenceZero(std::vector<int> numbers){
    for (int i = 0; i < numbers.size()-1; i++)
    {
        if (numbers[i] + numbers[i+1]==0)
        {
            return true;
        }
    }
    return false;
}

File: PythonLectures/HomeworkPy02/Task2.py
# 2.Generator to Yield Fibonacci Sequence
# Write a generator function that yields the Fibonacci sequence up to n numbers.
# Input: 5
# Output: 0, 1, 1, 2, 3
def fibonacci(n):
    iteration = 0
    num1 = 0
    num2 = 1
    while iteration < n:
        yield str(num1)
        num2 = num1 + num2
        num1 = num2 - num1
        iteration += 1


numInput = int(input())
result = str()
for testNum in fibonacci(numInput):
    result += str(testNum) + ', '
result = result[:-1:]
print(result)
